### PR TITLE
TS-4976: Regularize plugins - redirect_1

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
This actually needs a serious review. This hasn't been built for a long time, it used a stat API so old I had a hard time finding any reference to it, so I'm not confident I've extracted the relevant features. In addition there is the question if any of this should be used in the statistics documentation section.

The statistics example was added to this PR, along with documentation updates for statistics involving both the statistics example plugin and the redirect example plugin.

### CODE REVIEW

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/211/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/197/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/193/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1892/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/322/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1783/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/455/

File: example/redirect_1/redirect_1.c [hunk 1]
```diff
@@ -41,33 +41,27 @@
 
 #include <ts/ts.h>
 
+#define PLUGIN_NAME "redirect_1"
+
 static in_addr_t ip_deny;
 
 /*
  * uncoupled statistics variables:
```
Comment (maskit) [inline example/redirect_1/redirect_1.c]: There's no coupled statistic anymore, so I think we can remove "uncoupled".

**Comment by SolidWallOfCode** [comment]
@maskit - Should the plugin stats have a prefix? I think that has been discussed but I don't recall off hand what the conclusion was. I think it was that the stats should be in the "plugin." prefix to avoid collisions.

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/212/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/226/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1907/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/208/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/337/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1798/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/470/

**Comment by maskit** [comment]
@SolidWallOfCode Yes, I think they should, I don't remember the discussion though. I think having a prefix is good idea, and the plugin should be a role model since it is an example.

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/266/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/280/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/262/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1961/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/391/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1853/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/524/

**Comment by SolidWallOfCode** [comment]
I was leaving them unsquashed while I was still working on them.

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/280/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/294/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/276/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1975/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1867/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/539/

**Pull request merged**